File: Soldier.php
<?php

namespace Army;

use Weapon\Weapon;

class Soldier
{
    private $knife;
    private $sapperBlade;
    private $automatic;
    private $pistol;
    private $weaponPrice;

    final public function setKnife(Weapon $knife)
    {
        $this->knife = $knife;
        $this->weaponPrice += $knife->getPrice();
    }

    final public function setPistol(Weapon $sapperBlade)
    {
        $this->sapperBlade = $sapperBlade;
        $this->weaponPrice += $sapperBlade->getPrice();
    }

    final public function setAutomatic(Weapon $automatic)
    {
        $this->automatic = $automatic;
        $this->weaponPrice += $automatic->getPrice();
    }

    final public function setSapperBlade(Weapon $pistol)
    {
        $this->pistol = $pistol;
        $this->weaponPrice += $pistol->getPrice();
    }

    final public function getWeaponPrice()
    {
        return $this->weaponPrice;
    }

    final public function getAutomatic()
    {
        return $this->automatic;
    }

    final public function getPistol()
    {
        return $this->pistol;
    }

    final public function getKnife()
    {
        return $this->knife;
    }

    final public function getSapperBlade()
    {
        return $this->sapperBlade;
    }
}

File: Soldier.py
from Knife import Knife
from SapperBlade import SapperBlade
from Automatic import Automatic
from Pistol import Pistol


class Soldier:
    __knife = None
    __sapperBlade = None
    __automatic = None
    __pistol = None
    __weaponPrice = None

    def __init__(self, name):
        self.__name = name

    @property
    def name(self):
        return self.__name

    @property
    def knife(self):
        return self.__knife

    @knife.setter
    def knife(self, value: Knife):
        self.__knife = value

    @property
    def sapper_blade(self):
        return self.__sapperBlade

    @sapper_blade.setter
    def sapper_blade(self, value: SapperBlade):
        self.__sapperBlade = value

    @property
    def automatic(self):
        return self.__automatic

    @automatic.setter
    def automatic(self, value: Automatic):
        self.__automatic = value

    @property
    def pistol(self):
        return self.__pistol

    @pistol.setter
    def pistol(self, value: Pistol):
        self.__pistol = value

    @property
    def weapon_price(self):
        result = 0
        if self.__knife is not None:
            result += self.__knife.price
        if self.__pistol is not None:
            result += self.__pistol.price
        if self.__automatic is not None:
            result += self.__automatic.price
        if self.__sapperBlade is not None:
            result += self.__sapperBlade.price

        return result
